public/app/plugins/datasource/loki/result_transformer.ts

```diff
@@ -127,6 +130,9 @@ export function appendResponseToBufferedData(response: LokiTailResponse, data: M
   const labelsField = data.fields[3];
   const idField = data.fields[4];
 
+  // We need to store and track all used uids to ensure that uids are unique
+  const usedUids: { string?: number } = {};
```

Comment:
This won't probably work the same way as in `lokiStreamResultToDataFrame` as this is called only on the new lines. We probably have take all the current ids in `data` an use that as a map, otherwise we would be checking here only the few new lines that came later. I mean realistically this could be a problem only if the same line came in 2 separate batches from the websocket not sure how likely that is. What will happen in case we have 2 same ids when livestreaming? If it's only the visual fade highlight maybe I would not just add a comment here that this could happen but would not complicate the code right now.